src/Razor/test/Microsoft.VisualStudio.Razor.IntegrationTests/InProcess/EditorInProcess_Caret.cs

```diff
@@ -27,6 +27,33 @@ public async Task MoveCaretAsync(int position, CancellationToken cancellationTok
             view.Caret.MoveTo(point);
         }
 
+        public async Task WaitForCaretMoveAsync(CancellationToken cancellationToken)
```

Comment:
I thought we already had this one. Now that I think about it, this is _technically_ a bit dangerous right? If the caret manages to move before we call this function and set PositionChanged callback the caret won't move again and we'll just wait for timeout. Maybe this should be consumed as:
```c#
var caretWaitTask = WaitForCaretMoveAsync(token);
await TestServices.Editor.InvokeGoToDefinitionAsync(HangMitigatingCancellationToken);
await caretWaitTask;
```
? Does that actually work? Maybe there's a better model? Also, this is probably a problem in the other spots that use this model too.